hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/controller/ContractController.java

```diff
@@ -35,23 +41,66 @@
 import org.springframework.web.bind.support.WebExchangeBindException;
 import reactor.core.publisher.Mono;
 
+import com.hedera.mirror.web3.exception.InvalidTransactionException;
+import com.hedera.mirror.web3.service.ContractCallService;
+import com.hedera.mirror.web3.service.model.CallServiceParameters;
 import com.hedera.mirror.web3.viewmodel.ContractCallRequest;
 import com.hedera.mirror.web3.viewmodel.ContractCallResponse;
 import com.hedera.mirror.web3.viewmodel.GenericErrorResponse;
+import com.hedera.services.evm.store.models.HederaEvmAccount;
 
 @CustomLog
 @RequestMapping("/api/v1/contracts")
+@RequiredArgsConstructor
 @RestController
 class ContractController {
 
     static final String NOT_IMPLEMENTED_ERROR = "Operation not supported yet!";
+    private final ContractCallService contractCallService;
 
     @PostMapping(value = "/call")
     Mono<ContractCallResponse> call(@RequestBody @Valid ContractCallRequest request) {
-        throw new UnsupportedOperationException(NOT_IMPLEMENTED_ERROR);
+        if (request.isEstimate()) {
+            throw new UnsupportedOperationException(NOT_IMPLEMENTED_ERROR);
+        }
+        //make sure there is a valid sender for value transfers
+        if (request.getValue() > 0 && request.getFrom() == null) {
```

Comment:
Can we move these lines into `ContractCallRequest.isValidValueTransfer()`
```suggestion
        if (!request.isValidValueTransfer())
```